db/db_test.cc

```diff
@@ -3981,6 +3981,9 @@ class MockedRateLimiterWithNoOptionalAPIImpl : public RateLimiter {
 
   ~MockedRateLimiterWithNoOptionalAPIImpl() override {}
 
+  const char* Name() const override {
+    return "MockedRateLimiterWithNoOptionalAPI";
+  }
```

Comment:
Just make sure I understand - with this PR change, subclasses of RateLimiter will have to implement `const char* Name() const` (like what you did with `MockedRateLimiterWithNoOptionalAPI`) otherwise there will be a compilation error. Am I correct?

Reply:
I pushed a change that will allow (until deprecated) RateLimiter classes with an empty Name().

> Thank you for working on this! As I am still pretty new to the Customizable project, I will appreciate if there is a bit more on "why we are doing this" **specifically** for RateLimiter, even though we learnt about the general benefit from @mrambacher's sharing.
> 
> The reason why I'd like to know is that this PR has quite some codes to merge into the codebase and it would be nice to justify it with a usecase/need/issue at the time it is merged in, in my opinion. Maybe there is already such a usecase/need/issue that I am not aware of - if that's the case, feel free to refer to them as I'd love to know!
> 
> Thanks!

As it stands now, there are a set of classes (RateLimiter among them) that cannot be serialized and re-created from the Options file.  Additionally, custom classes cannot easily be implemented and substituted, since they are not plugged into the framework.

One of my goals was to complete by year-end making all of the classes that allow custom implementations from the Options to be Customizable.  RateLimiter is one of the remaining hold-outs.  The change does not increase the complexity of the code or cause a performance degradation but does make the class on the same footing as the other Options classes.

Reply:
Sorry for responding late but thanks for the context and the new no name test!